File: alds1/1d_maximum_profit.c
#define dbg(fmt,...) fprintf(stderr,fmt,__VA_ARGS__)
#define dpri(x) dbg(#x ": %d\n", x)
#define dprs(x) dbg(#x ": %s\n", x)
#include <stdio.h>
typedef long long ll;
const int MYINF = 1e9+7;

int main(int argc, char **argv)
{
	int i, n, r, rmin, prof;

	scanf("%d", &n);
	scanf("%d", &rmin);
	scanf("%d", &r);
	prof = r - rmin;
	if (rmin > r) rmin = r;

	for (i = 2; i < n; i++) {
		scanf("%d", &r);
		if ((r - rmin) > prof) {
			prof = r - rmin;
		}
		if (r < rmin) rmin = r;
	}
	printf("%d\n", prof);

	return 0;
}

File: alds1/1d_maximum_profit.go
package main

import "fmt"

func max(a, b int) int {
	if a > b {
		return a
	} else {
		return b
	}
}

func min(a, b int) int {
	if a < b {
		return a
	} else {
		return b
	}
}

func main() {
	var n int
	fmt.Scanf("%d", &n)
	var minv int
	fmt.Scanf("%d", &minv)
	maxd := -1 << 30
	for i := 1; i < n; i++ {
		var v int
		fmt.Scanf("%d", &v)
		maxd = max(maxd, v-minv)
		minv = min(minv, v)
	}
	fmt.Println(maxd)
}
